packages/shared/lib/clients/orchestrator.ts

```
@@ -117,13 +118,16 @@ export class Orchestrator {
             'connection.provider_config_key': connection.provider_config_key,
             'connection.environment_id': connection.environment_id
         };
+        if (!connection.id) {
+            throw new NangoError('invalid_input', { connection });
```

this is going to throw the error instead of returning an Err

---

See line 242 in this file. This function has a big try/catch and throws errors above which then get caught and returned as Err. 😬

---

Fixed in a39804e00